tritium-metrics-jvm/src/main/java/com/palantir/tritium/metrics/jvm/JvmMetrics.java

```diff
@@ -80,6 +81,10 @@ private static void registerAttributes(InternalJvmMetrics metrics) {
                 .javaRuntimeVersion(System.getProperty("java.runtime.version", "unknown"))
                 .javaVendorVersion(System.getProperty("java.vendor.version", "unknown"))
                 .javaVmVendor(System.getProperty("java.vm.vendor", "unknown"))
+                .enablePreview(
+                        runtimeMxBean.getInputArguments().contains("--enable-preview")
```

Comment:
Seems like we don't have junit tests to prove this method gives us sensible values, but I did validate this:

```
$ jshell --enable-preview
|  Welcome to JShell -- Version 15.0.7
|  For an introduction type: /help intro

jshell> java.lang.management.ManagementFactory.getRuntimeMXBean().getInputArguments()
$1 ==> [-agentlib:jdwp=transport=dt_socket,address=localhost:61023, --enable-preview]
```

Reply:
I don't think we _ever_ want to --enable-preview for the compilation of published jars, because they can _only_ be run by the same java version that was used to compile them (i.e. if we did this with java15 for `:tritium-metrics-jvm` then anyone running java17 in production would fail to start)... I think this would quickly lead to unsolveable dependency graphs and much angst in #dev-backend-infra.

We'd need to either invent a gradle API for setting --enable-preview on the test sourceset only (I didn't add support for this initially https://github.com/palantir/gradle-baseline/pull/2322), or make up an entire separate gradle project that we would enable preview on.